tests/test_dns.py

```diff
@@ -69,8 +69,11 @@ def test_from_cache(self):
 
         service_name = "test_service.examle.com"
 
+        entry = Mock(spec_set=["expires"])
+        entry.expires = 999999999
+
         cache = {
-            service_name: [object()]
+            service_name: [entry]
         }
 
         servers = yield resolve_service(
```

Comment:
Now that resolve_service takes a clock, it might be worth using a mock clock here?
Also might be nice to have a test that checks that things expire.